### PR TITLE
1102: Document minimum required JDK version 16 for Skara

### PR DESCRIPTION
**Pull request by erikj79**
Change documentation to reflect that Skara now requires JDK 16 to build.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [SKARA-1102](https://bugs.openjdk.java.net/browse/SKARA-1102): Document minimum required JDK version 16 for Skara

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Iris Clark](https://openjdk.java.net/census#iris) (@irisclark - no project role) ⚠️ Review applies to 35fa6b8c76c0870a8fbf1f1167040d03995ef16d

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/skara pull/1195/head:pull/1195` \
`$ git checkout pull/1195`

Update a local copy of the PR: \
`$ git checkout pull/1195` \
`$ git pull https://git.openjdk.java.net/skara pull/1195/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1195`

View PR using the GUI difftool: \
`$ git pr show -t 1195`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/skara/pull/1195.diff">https://git.openjdk.java.net/skara/pull/1195.diff</a>

</details>

### CODE REVIEW

**Review by kevinrushforth — APPROVED** [review]
You might also want to change the example of `openjdk-15.0.1_osx-x64_bin/jdk-15.0.1.jdk` to `openjdk-16.0.1_osx-x64_bin/jdk-16.0.1.jdk`, but otherwise looks good.

**Review by kevinrushforth — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -46,8 +46,8 @@ external Git source code hosting providers are available:
 
 ## Building
 
-[JDK 14](http://jdk.java.net/14/) or later and [Gradle](https://gradle.org/)
-6.6 or later is required for building. To build the project on macOS or
+[JDK 16](http://jdk.java.net/16/) or later and [Gradle](https://gradle.org/)
```
Comment (irisclark) [inline README.md:49]: How often is the minimum JDK version for Skara updated?  If it's always the last GA'd version of the JDK, perhaps that sort of phrasing should be used instead?

Comment (irisclark) [reply README.md:49]: Thanks for the explanation:  https://mail.openjdk.java.net/pipermail/skara-dev/2021-July/005097.html

I agree that the JDK version needs to remain explicit.

**Review by irisclark — APPROVED** [review]

**Comment by erikj79** [comment]
Added 'release 16' for all Java compilation in Gradle. Verified that setting it to 15 triggered the same compilation errors I saw in Intellij when setting it to 15 there.

**Review by kevinrushforth — APPROVED** [review]
Good idea to set `--release 16`

**Pull request closed without merging**